rama-dns/src/lib.rs

```diff
@@ -79,6 +79,89 @@ impl<R: DnsResolver<Error: Into<BoxError>>> DnsResolver for Option<R> {
     }
 }
 
+#[derive(Debug)]
```

Comment:
I think it might be a good idea to move all your new code here to a new file: `rama-dns/src/chain.rs`. Can be a private fail of which you pub use (re-export) types here.